sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -298,10 +298,9 @@ protected AmqpConnectionScope()
         /// <param name="receiveMode">The <see cref="ServiceBusReceiveMode"/> used to specify how messages are received. Defaults to PeekLock mode.</param>
         /// <param name="sessionId">The session to connect to.</param>
         /// <param name="isSessionReceiver">Whether or not this is a sessionful receiver.</param>
+        /// <param name="isProcessor">Whether or not the receiver is being created for a processor.</param>
```

Comment:
We're starting to accumulate a few of these `isBlah` members.   Should we consider some kind of `ClientType` enumeration to make it easier to extend in the future?

Reply:
I removed a lot of the usages of these flags by storing off instance fields on the AmqpReceiver.